src/components/RegisterForm.test.tsx

```diff
@@ -0,0 +1,205 @@
+import { render, screen, within, fireEvent } from '@testing-library/react'
```

Comment:
Good tests! 🚀  I think we can shrink the code down a little if we use helper functions to reduce code duplications: See comments.